asv_bench/benchmarks/groupby.py

```diff
@@ -510,6 +512,33 @@ def time_groupby_extra_cat_nosort(self):
         self.df_extra_cat.groupby("a", sort=False)["b"].count()
 
 
+class CategoricalFrame:
+    # benchmark grouping with operations on categorical values (GH #32976)
+    param_names = ["groupby_type", "value_type"]
```

Comment:
oarameterize over 

mean and head as well

i would also reduce the number of groups as well 
it will still show an appreciable diff

Reply:
done. but I swapped mean for count since mean requires numeric types

Reply:
i also removed the str groupby parameter option to limit the number of tests but the benchmark results are the same either way for this diff.

here are the latest results:
```
       before           after         ratio
     [22cf0f5d]       [a16f6a23]
     <master>                   
-         167±1ms      2.72±0.09ms     0.02  groupby.CategoricalFrame.time_groupby(<class 'int'>, <class 'str'>, 'last')
-         168±2ms      2.66±0.07ms     0.02  groupby.CategoricalFrame.time_groupby_ordered(<class 'int'>, <class 'str'>, 'last')
-         174±2ms      2.65±0.09ms     0.02  groupby.CategoricalFrame.time_groupby_ordered(<class 'int'>, <class 'int'>, 'last')
-         175±2ms      2.61±0.05ms     0.01  groupby.CategoricalFrame.time_groupby(<class 'int'>, <class 'int'>, 'last')

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE INCREASED.
```